build/lifecycle.go

```diff
@@ -50,16 +57,34 @@ type LifecycleOptions struct {
 }
 
 func (l *Lifecycle) Execute(ctx context.Context, opts LifecycleOptions) error {
-	cacheImage := cache.New(opts.Image, l.docker)
-	l.logger.Verbose("Using cache image %s", style.Symbol(cacheImage.Image()))
+	l.Setup(opts)
+	defer l.Cleanup()
+
+	lifecycle020OrLater := l.hasVersion020OrLater()
+	var buildCache, launchCache Cache
+	if lifecycle020OrLater {
+		buildCache = cache.NewVolumeCache(opts.Image, "build", l.docker)
+		launchCache = cache.NewVolumeCache(opts.Image, "launch", l.docker)
+		l.logger.Verbose("Using build cache volume %s", style.Symbol(buildCache.Name()))
+		l.logger.Verbose("Using launch cache volume %s", style.Symbol(launchCache.Name()))
```

Comment:
I wonder if exposing the idea of the launch cache to users is confusing (especially in comparison to build cache). I'm not sure they necessarily need to know about it

Reply:
Agreed. We'll remove it.